doc/ruleSets/PhpCsFixerRisky.rst

```diff
@@ -31,6 +31,7 @@ Rules
 - `no_unset_on_property <./../rules/language_construct/no_unset_on_property.rst>`_
 - `php_unit_data_provider_name <./../rules/php_unit/php_unit_data_provider_name.rst>`_
 - `php_unit_data_provider_return_type <./../rules/php_unit/php_unit_data_provider_return_type.rst>`_
+- `php_unit_data_provider_static <./../rules/php_unit/php_unit_data_provider_static.rst>`_
```

Comment:
:-1:

This is not about style, this is about migration. 
All migration rules shall be part of PHPUnit migration ruleset.

I would rather recommend projects to keep config like following:
```
[
    '@PHP{xy}Migration' => true,
    '@PHPUnit{xy}Migration' => true,
    '@PhpCsFixer' => true, // or @Symfony, @PSR, ...
]
```

if there is a non-migration rule, it could be part of vendor style (eg `@Fixer...`, `@Symfony`)

Reply:
Even if static providers make sense in all PHPUnits?

Reply:
Were static data providers always available? If yes, then I would consider it as style rule rather than migration rule. But if they were introduced in some specific version, then it surely is migration rule 😉. Personally: I don't know 😅.

Reply:
They are supported since "always": https://github.com/sebastianbergmann/phpunit/blob/4.8.36/src/Util/Test.php#L440